src/pages/addMoviePage/AddMoviePage.tsx

```diff
@@ -2,19 +2,75 @@ import React from "react";
 import cross from "./../../images/x.svg"
 import AddMovieInput from "./components/AddMovieInput";
 import "./addMoviePage.css"
+import {formInputsObj, movieValidationSchema} from "./formInputList";
+import {Formik, Form, Field} from "formik";
+import {useDispatch} from "react-redux";
+import {putMoviesThunk} from "../../store/movies-store/putMovies.thunk";
+
+
+const initialValues = {
+  title: '',
+  release_date: '',
+  poster_path: '',
+  genres: [],
+  overview: '',
+  runtime: 0,
+  budget: 0,
+  revenue: 0,
+  tagline: '',
+  vote_average: 0,
+  vote_count: 0
+};
+
 
 
 const AddMoviePage = ({active, setActive}: {active: boolean; setActive: (status: boolean) => void}) => {
+  const dispatch = useDispatch();
+
+  const submitForm = (values: any) => {
+    const data = {
+      ...values,
+    }
+    dispatch(putMoviesThunk(data));
+  }
+
   return (
       <div className={active ? "modal active" : "modal"} onClick={() => setActive(false)}>
        <div className="addMoviePageContainer" onClick={e => e.stopPropagation()}>
         <img className="crossImg" src={cross} onClick={() => setActive(false)}/>
-        <AddMovieInput type="text" placeholder="Title" label="Title"/>
-        <AddMovieInput type="date" placeholder="Select Date" label="Release Date"/>
-        <AddMovieInput type="text" placeholder="Movie URL" label="Movie URL here"/>
-        <AddMovieInput type="text" placeholder="Genre" label="Select Genre"/>
-        <AddMovieInput type="text" placeholder="Overview" label="Overview here"/>
-        <AddMovieInput type="text" placeholder="Runtime" label="Runtime here"/>
+        <Formik  initialValues={initialValues} onSubmit={submitForm}  validationSchema={movieValidationSchema}>
+          {
+            () => (
+              <Form onChange={event => console.log('event: ', event)}>
+                {Object.keys(formInputsObj).map((key) => {
+                  return (
+                    <React.Fragment key={key}>
```

Comment:
Do we need `Fragment` here? Also, `key` prop should be assigned only for the parent wrapper. Please,  check lines 47, 48

Reply:
<Field /> cannot get a key prop and I don't want to wrap <Field /> in any HTML elements. so this is the reason why I choose to use <React.Fragment />, also, Fragment is able to receive key check docs - https://reactjs.org/docs/fragments.html